src/services/api/external-id.ts

```diff
@@ -0,0 +1,68 @@
+import tenants from '../../tenants'
+import OneBlinkAppsError from '../errors/oneBlinkAppsError'
+import { HTTPError, postRequest } from '../fetch'
+
+async function generateExternalId(formId: number, abortSignal?: AbortSignal) {
+  const url = `${tenants.current.apiOrigin}/forms/${formId}/receipt-id`
+  try {
+    const { externalId } = await postRequest<{ externalId: string }>(
+      url,
+      {},
+      abortSignal,
+    )
+    return externalId
+  } catch (err) {
+    console.warn(
+      'Error occurred while attempting to generate External ID from API',
+      err,
+    )
+    if (err instanceof OneBlinkAppsError) {
+      throw err
+    }
+
+    const error = err as HTTPError
+    switch (error.status) {
+      case 401: {
+        throw new OneBlinkAppsError(
+          'You cannot submit this form until you have logged in. Please login and try again.',
+          {
+            originalError: error,
+            httpStatusCode: error.status,
+            requiresLogin: true,
+          },
+        )
+      }
+      case 403: {
+        throw new OneBlinkAppsError(
+          'You do not have access to this form. Please contact your administrator to gain the correct level of access.',
+          {
+            originalError: error,
+            httpStatusCode: error.status,
+            requiresAccessRequest: true,
+          },
+        )
+      }
+      case 400:
```

Comment:
Can we please move this one into its own `case` and use the `error.message` as the message.